test/e2ev2/tmanager/chain.go

```diff
@@ -0,0 +1,234 @@
+package tmanager
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	sdkmath "cosmossdk.io/math"
+	appkeepers "github.com/babylonlabs-io/babylon/v4/app/keepers"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/server"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/stretchr/testify/require"
+
+	appparams "github.com/babylonlabs-io/babylon/v4/app/params"
+	"github.com/babylonlabs-io/babylon/v4/testutil/datagen"
+)
+
+const (
+	CHAIN_ID_BABYLON = "bbn"
+	CHAIN_ID_BSN     = "consumer-bsn"
+)
+
+var (
+	BabyInitialBalance = sdkmath.NewInt(1_000_000_000000) // 1kk ubbn
+)
+
+// ChainConfig defines configuration for a blockchain
+type ChainConfig struct {
+	ChainID               string
+	Home                  string
+	ValidatorCount        int
+	NodeCount             int
+	BlockTime             time.Duration
+	EpochLength           int64
+	VotingPeriod          time.Duration
+	ExpeditedVotingPeriod time.Duration
+	BTCConfirmationDepth  int
+	GasLimit              int64
+}
+
+// Chain represents a blockchain with multiple nodes
+type Chain struct {
+	Nodes          []*Node
+	Validators     []*ValidatorNode
+	Tm             *TestManager
+	InitialGenesis *InitGenesis
+	Config         *ChainConfig
+}
+
+// NewChainConfig creates a new chain configuration with default values
+func NewChainConfig(tempDir, chainID string) *ChainConfig {
+	return &ChainConfig{
+		ChainID: chainID,
+		Home:    filepath.Join(tempDir, chainID),
+		// starts with 2 nodes (1 val, one non-validator)
+		ValidatorCount:        1,
+		NodeCount:             1,
+		BlockTime:             5 * time.Second,
```

Comment:
can we use lower block time (eg 1 or 2 sec) so tests don't take too long?

Reply:
move to 2s